### PR TITLE
Docker Integration by adding dockerfile and docker.md

### PR DESCRIPTION
**Pull request by liu-ziyang**
# Description
Add docker file for docker builds

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# References
https://github.com/napari/napari/issues/903

# How has this been tested?
- all tests pass with my change

## Final checklist:
- [X] My PR is the minimum possible work for the desired functionality
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Czaki** [comment]
Which libraries are missed? 
Maybe this list will help https://pytest-qt.readthedocs.io/en/latest/troubleshooting.html#github-actions

or method from this comment allow to find missed one https://github.com/pytest-dev/pytest-qt/issues/266#issuecomment-510035926

**Comment by sofroniewn** [comment]
> Which libraries are missed?

Sorry @Czaki I'm not quite sure what you mean here, can you elaborate?

@ziyangczi is there any follow up here, if not I will merge asap, sorry for the delay!!

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ENV DEBIAN_FRONTEND noninteractive
+
+# install graphical libraries used by qt and vispy
+RUN apt-get install -qqy libxi6 libglib2.0-0 fontconfig libgl1-mesa-glx libfontconfig1 libxrender1 libdbus-1-3
+
+# install napari release version
+RUN pip3 install napari[all]
+
+# library missing when using provided pyqt version, reversed to an earlier version for now
+RUN pip3 install PyQt5==5.11.3
```
Comment (Czaki) [inline dockerfile]: @sofroniewn I think about this comment in Dockerfile

Comment (liu-ziyang) [reply dockerfile]: I think something was fixed along with 0.3.8, I was able to build with this line removed, updating now

**Comment by sofroniewn** [comment]
We're getting an error on checking of the manifest

```
check-manifest
lists of files in version control and sdist do not match!
missing from sdist:
  dockerfile
suggested MANIFEST.in rules:
  include dockerfile
```

we probably don't want the dockerfile in there (right @jni?) so could just do
```
exclude dockerfile
```
and then the test will be happy I think

**Comment by jni** [comment]
> we probably don't want the dockerfile in there (right @jni?)

Right! It would be pretty weird to ship that in the sdist. 😂

**Review by sofroniewn — APPROVED** [review]
Ok, this looks good to me! Let's get it in there, we can continue to update if we need / if people make requests!! I've also fixed the manifest.

**Comment by johschmidt42** [comment]
Any progress on a working dockerfile for linux OS? I'd be satisfied with a nvidia-docker image.

**Comment by sofroniewn** [comment]
> Any progress on a working dockerfile for linux OS? I'd be satisfied with a nvidia-docker image.

I believe the latest was merged here and the most up-to-date resource we have on napari and docker is here https://napari.org/guides/stable/docker.html. @johschmidt42 if you'd like to contribute anything to improve the situation it would be more than welcome, or if anyone else reading has any ideas too that would be great. I'm afraid I don't know any more than this

**Comment by johschmidt42** [comment]
I've tried the most recent and past docker images but none of them worked for me with X11 forwarding on a Linux machine. I've spent a day to figure out why, built my own docker images to make it run, but no success. Surprisingly, running a X Window server on a windows machine as suggested in the docs actually works but is far from desireable because it is painfully slow..
Running a simple qt app with python works in a docker container with x window forwarding but fails with napari due to driver issues and opengl. There are nvidia-OpenGL docker images that should theoretically make it work if you have a nvidia card but that also wasn't successful. So I'd really appreciate it if someone figures out how to make it run..

**Pull request merged**